src/features/accessibility.py

```diff
@@ -1,77 +1,86 @@
 import asyncio
 import json
+from concurrent.futures import Executor
 
 from aiohttp import ClientSession
+from pydantic import BaseModel
 
 from app.models import Explanation, StarCase
-from core.extractor import NO_EXPLANATION, MetadataBase
+from core.extractor import Extractor
 from core.website_manager import WebsiteData
-from lib.constants import ACCESSIBILITY
 from lib.settings import LIGHTHOUSE_TIMEOUT, LIGHTHOUSE_URL
 
 _DESKTOP = "desktop"
 _MOBILE = "mobile"
+_ACCESSIBILITY = "accessibility"
 
 _LIGHTHOUSE_SCORE_SUITABLE = "Lighthouse accessibility score is suitable"
 _LIGHTHOUSE_SCORE_TOO_LOW = "Lighthouse accessibility score is too low"
 
 
-@MetadataBase.with_key()
-class Accessibility(MetadataBase):
+class AccessibilityScores(BaseModel):
+    """
+    Expose the lighthouse accessibility scores as extractor extra data.
+    Scores range from zero to one, higher is better.
+    """
 
-    star_levels = [0.7, 0.8, 0.85, 0.9, 0.95, 1]
+    mobile_score: float
+    desktop_score: float
+    average_score: float
 
-    async def _execute_api_call(
-        self,
-        website_data: WebsiteData,
-        session: ClientSession,
-        strategy: str = "desktop",
-    ) -> float:
-        params = {
-            "url": website_data.url,
-            "category": ACCESSIBILITY,
-            "strategy": strategy,
-        }
-        container_url = f"{LIGHTHOUSE_URL}/{ACCESSIBILITY}"
 
-        response = await session.get(url=container_url, timeout=LIGHTHOUSE_TIMEOUT, json=params)
+class Accessibility(Extractor[AccessibilityScores]):
+    key = _ACCESSIBILITY
 
-        if response.status == 200:
-            # expected result looks like {"score": [0.123]}
-            return float(json.loads(await response.text())["score"][0])
-        else:
-            raise Exception(f"Request to lighthouse failed with {response.status}: {await response.text()}")
+    def __init__(self, lighthouse_timeout: int = LIGHTHOUSE_TIMEOUT, lighthouse_url: str = LIGHTHOUSE_URL):
+        """
+        :param lighthouse_timeout: The timeout in seconds for requests issued to the lighthouse service.
+        :param lighthouse_url: The url of the lighthouse service.
+        """
+        self.timeout = lighthouse_timeout
+        self.url = lighthouse_url
```

Comment:
In _execute_api_call there is another "url", which are not the same. I suggest to rename this, e.g., lighthouse_uri? Otherwise we have two urls later in one function :/

Reply:
renamed the url and timeout member variable